core/src/main/java/org/moditect/commands/GenerateModuleInfo.java

```diff
@@ -234,13 +245,30 @@ private void updateModuleInfo(Map<String, Boolean> optionalityPerModule, ModuleD
         }
 
         for (String usedService : uses) {
-            moduleDeclaration.getDirectives().add( new ModuleUsesDirective(parseName(usedService)) );
+            moduleDeclaration.getDirectives().add( new ModuleUsesDirective(JavaParser.parseName(usedService)) );
         }
 
+        provides.stream().map(
+                providedService -> providedService.split("with")
```

Comment:
Would a service named `FoowithBar` make things fall apart? Better to split along `" with "` (i.e. with surounding whitespace) perhaps?

Reply:
Alternatively, it could be structured differently in the incoming configuration:

```
<provides>
    <service>org.test.spi.serviceProvider</service>
    <implementation>
        <implementation>org.test.spi.serviceProviderImpl1</implementation>
        <implementation>org.test.spi.serviceProviderImpl2</implementation>
    </implementation>
</provides>
```

More verbose of course. I'm open for both ways.

Reply:
Thanks for your comments :)

I think that we should give the choice to do it 3 ways:
- inline content (<provides>org.test.spi.serviceProvider with org.test.spi.serviceProviderImpl1, org.test.spi.serviceProviderImpl2)
- simple structure (<provides><service>org.test.spi.serviceProvider</service><implementation>org.test.spi.serviceProviderImpl</implementation><provides>)
- composite structure (<provides><service>org.test.spi.serviceProvider</service><implementations><implementation>org.test.spi.serviceProviderImpl1</implementation></implementations><provides>)

I'm not sure whether it's something easy to do. I'll look at this last bit later.

Reply:
For the sake of simplicity, there should be a single way for doing this. Inline is fine for me, if it's your preferred approach. Just let's make sure a service name containing "with" won't break things.

Reply:
Ok, I've just updated the file readme.md with the inline version. Although I understand that being able to do a more structured declaration should be a good thing, I'm not really motivated to actually implement it :p